src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -1412,6 +1412,102 @@ cudnnSetOpTensorDescriptor(opTensorDesc,
                 writer.block_end();
             }
 
+            template <>
+            void GPU_Emitter::EMITTER_DECL(ngraph::op::BatchNorm)
+            {
+                const ngraph::op::BatchNorm* batchnorm =
+                    static_cast<const ngraph::op::BatchNorm*>(node);
+
+                auto& cudnn_emitter =
+                    external_function->get_primitive_emitter()->get_cudnn_emitter();
+
+                CUDNNEmitter::Prop dir;
+                if (batchnorm->get_training_flag() && args.size() == 3)
+                {
+                    // forward train
```

Comment:
I'm all for comments, but does this comment add anything? If `dir` were renamed `direction` then it would be even more clear. The best code has no comments in it because the intent is clear.

Reply:
Completely agreed, that is what I was going for with the enum, the comments were left over from before. I also have renamed dir->direction as it is better that way.